app/models/manageiq/providers/foreman/configuration_manager.rb

```diff
@@ -31,11 +37,30 @@ def self.description
     @description ||= "Foreman Configuration".freeze
   end
 
+  def provider
+    super || ensure_provider
+  end
+
   def image_name
     "foreman_configuration"
   end
 
   def self.display_name(number = 1)
     n_('Configuration Manager (Foreman)', 'Configuration Managers (Foreman)', number)
   end
+
+  def ensure_provider
+    build_provider
+
+    provider.configuration_manager = self
+    provider.name = name
+    provider.zone = zone
+
+    provider
+  end
+
+  def ensure_provider_name_and_zone
+    provider.name = name.split(" Configuration Manager").first
```

Comment:
The `split` isn't the safest choice as it can mess up the provider name if the `Configuration Manager` isn't a suffix. I'd rather go with `.sub(/ Configuration Manager$/)` instead.

Reply:
The `Provider#ensure_managers` ensures that there will be a `Configuration Manager` suffix but sure I can use `.sub`